dace/frontend/python/replacements.py

```diff
@@ -527,30 +535,43 @@ def _arange(pv: ProgramVisitor, sdfg: SDFG, state: SDFGState, *args, **kwargs):
     else:
         start, stop, step = args
 
+    if isinstance(start, str):
```

Comment:
The underlying issue here is data-dependent array shape?

Reply:
Yes, though generally solvable with a symbol assignment and another state, for now at least symbols and constants should be supported.